preprocessors/openstreetmap/osm_service.py

```diff
@@ -154,6 +174,346 @@ def process_streets_data(OSM_data, bbox_coordinates):
         return processed_OSM_data
 
 
+def get_new_nodes(unbounded_node_list, node_list, bbox_coordinates):
+    # node_list1 contains the unbounded nodes
+    # node_list  contains the bounded nodes
+    node_list1 = unbounded_node_list
+    i = len(node_list)
+    j = len(node_list1)
+
+    # Case I: When there is only one point (node) in the node_list,
```

Comment:
Case 1 and 2 can be written in one section. It seems like Case 1 is just a subcase of case 2.